backend/app/views/spree/admin/promotion_rules/create.js.erb

```diff
@@ -4,6 +4,12 @@ $('#rules .no-objects-found').hide();
 $('.product_picker').productAutocomplete();
 $('.user_picker').userAutocomplete();
 
+$('select.select2').select2({
```

Comment:
It doesn't seem the same to me with regard to what we're talking about. productAutocomplete and userAutocomplete above both target specific classes -- `data-` attributes would be better, but classes work -- rather than every single select2 on the page, which is what @jordan-brough was questioning right?

But this update in `create.js.erb` is weird to begin with (`.js.erb` is probably always troubling), it would be nice if there were a completely different way to do it. 

Meanwhile, I don't really understand why you need to set the select2 attributes `minimumResultsForSearch`, `dropdownAutoWidth`, and `allowClear` on ajax update here at all, but if you really do need to set them, and set them for every single select2 on the page, then this is the appropriate way to do it... but why is that going on?